### ISSUE TITLE
Keyboard combination Alt + arrows in the editor

### ISSUE THREAD

**Issue by raffaeler** [Microsoft/vscode#22364]
- VSCode Version: Code 1.10.2 (8076a19fdcab7e1fc1707952d652f0bb6c6db331, 2017-03-08T14:02:52.799Z)
- OS Version: Windows_NT ia32 10.0.14393
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|plantuml|jebbs|1.2.5|
|cpptools|ms-vscode|0.10.2|
|debugger-for-chrome|msjsdiag|2.6.0|
|language-vscode-javascript-angular2|nwallace|0.0.11|
|language-plantuml|qhoekman|0.0.3|
|nativescript|Telerik|0.6.1|;

---

Old schoold people like me, still want the numlock key disabled. This means that "ALT + arrows" (and using the arrows on the numpad of a desktop keyboard) results in inserting an ascii code instead in addition to the keyboard shortcut defined in VSCode.

Steps to Reproduce:

1. Open the editor with some text separated by newlines
2. ALT + Up arrow and ALT + Down arrow move the lines up and down
3. Now turn off the numlock on the keyboard
4. repeat the step at point (2):
   ALT + Up Arrow (on the numpad) lead to inserting the character ◘
   ALT + Down Arrow (on the numpad) lead to inserting the character ☻

I expect the behavior is consistent with Visual Studio where ascii characted are never injected on these keyboard combinations.

Thanks

**Comment by BartmanAbyss** [comment]
+1

**Comment by chienhsingwu** [comment]
Not just up and down. left at least will insert a character.

**Comment by GreenSliper** [comment]
really irritating

**Comment by DannyVarod** [comment]
It's been 6 years, any solution?

**Comment by tmsasaki** [comment]
I use a workaround for this issue:

* Install the MICROSOFT POWER TOYS (https://learn.microsoft.com/pt-br/windows/powertoys/).

* Use the KEYBOARD MANAGER to remap each ARROW NUMPAD KEY to the PHYSICAL NUMPAD KEY:

![image](https://github.com/microsoft/vscode/assets/83930/52eabee5-2277-4c50-9bee-93048b63b6f6)

Now ALT + NUMPAD ARROW is not inserting a character anymore.

**Comment by patfinder** [comment]
I think this is Windows keyboard behavior.  So it cannot/should not be "fixed" by VSCode.
This is call [Alt-code](https://en.wikipedia.org/wiki/Alt_code) input. This is **purposely used** to input special characters using it code.
For those who reach here by searching the way to input Unicode characters, use Word or Wordpad, type [Alt+Character number] to input the character, then you can copy it to VSCode or other editors that support Unicode.

**Comment by raffaeler** [comment]
> I think this is Windows keyboard behavior (Mac, Linux do have similar techniques). So it cannot/should not be "fixed" by VSCode.

This is true only if you have the numlock active. If you turn the numlock off, the behavior of VSCode is wrong.
In fact, if you repeat the same sequence on VS, it works as expected.

**Comment by DumbCounselor** [comment]
> I use a workaround for this issue (Windows only solution):
> 
> * Install the MICROSOFT POWER TOYS (https://learn.microsoft.com/pt-br/windows/powertoys/).
> * Use the KEYBOARD MANAGER to remap each ARROW NUMPAD KEY to the ARROW KEY:
> 
> ![image](https://github.com/microsoft/vscode/assets/83930/52eabee5-2277-4c50-9bee-93048b63b6f6)
> 
> Now ALT + NUMPAD ARROW is not inserting a character anymore.

Unfortunately didn't work for me. But autohotkey script worked:

```
NumpadIns::Ins
NumpadEnd::End
NumpadDown::Down
NumpadPgDn::PgDn
NumpadLeft::Left
NumpadRight::Right
NumpadHome::Home
NumpadUp::Up
NumpadPgUp::PgUp
NumpadDel::Del
```

**Comment by raffaeler** [comment]
BTW, I am now on Ubuntu 24.04. So please consider adopting a x-plat solution.

**Comment by DumbCounselor** [comment]
> BTW, I am now on Ubuntu 24.04. So please consider adopting a x-plat solution.

Guess the problem will never be fixed by editor.
On Windows i just remapped keys in OS with PowerToys. No problems since then.
May be for linux there are some alternatives.
